web/src/hooks/transaction/useSubmitTransaction.ts

```diff
@@ -133,31 +120,42 @@ const useSubmitTransaction = (
           transactionReceipt as TransactionReceipt,
           events as unknown as DecodeEventLogReturnType[],
         );
+        setIsLoading(false);
       }
     }
 
     if (error) {
       onError?.(error, rawError);
-      resetToUse();
+      setIsLoading(false);
+      resetBatchWriteContract();
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [transactionReceipt, isSuccess, isError]);
 
   return {
     onSubmitTransaction: () => {
-      if (!writeContractToUse && error) {
-        onError?.(error, rawError);
-        return;
-      }
-
-      if (options?.batchTx) {
-        writeContracts?.(contractCallConfig);
-      } else {
-        writeContract?.(contractCallConfig);
-      }
+      setIsLoading(true);
+      writeContracts?.(
+        {
+          contracts: contractCallConfig.contracts || [contractCallConfig],
+          capabilities: {
+            paymasterService: {
+              url: ZERODEV_PAYMASTER_URL,
+            },
+          },
+        },
+        {
+          // tx successfully sent to the bundler
+          onSuccess: () => {
+            if (typeof options?.onSent === 'function') {
+              options.onSent();
+            }
+          },
+        },
```

Comment:
Do we want to add onError logic here?

Reply:
that should already be handled by 
```
if (error) {
      onError?.(error, rawError);
      setIsLoading(false);
      resetBatchWriteContract();
    }
```

because any error here will also be set to `batchWriteContractError`